idea/idea-frontend-fir/src/org/jetbrains/kotlin/idea/frontend/api/fir/components/KtFirTypeProvider.kt

```diff
@@ -51,6 +58,30 @@ internal class KtFirTypeProvider(
         return type.asKtType()
     }
 
+    override fun getKtType(ktTypeReference: KtTypeReference): KtType? = withValidityAssertion {
+        when (val fir = ktTypeReference.getOrBuildFir(firResolveState)) {
+            is FirErrorTypeRef -> null
```

Comment:
There is `KtErrorClassType` in HL API. It's good to return it for the client too. So, the code may be simplified to `ktTypeReference.getOrBuildFirOfType<FirResolvedTypeRef>(firResolveState).coneType.asKtType()`

Reply:
Simplified as per the suggestion, thanks.

But, from UAST viewpoint, even such error type is converted to `PsiType` where cone error type is encoded. It should be `null`, though. This can be handled in UAST side, if the `KtType` from `KtTypeReference` is `KtClassErrorType`. I adjusted that part in FIR UAST PR.